chainer/datasets/concatenated_dataset.py

```diff
@@ -0,0 +1,38 @@
+class ConcatenatedDataset(object):
```

Comment:
How about implementing it with DatasetMixin? Handling indices will be much easier and nd.array index will be automatically supported.